node/src/components/consensus/era_supervisor/tests.rs

```diff
@@ -50,6 +50,107 @@ enum Event {
     ContractRuntime(ContractRuntimeRequest),
 }
 
+struct MockReactor {
+    scheduler: &'static Scheduler<Event>,
+}
+
+impl MockReactor {
+    fn new() -> Self {
+        MockReactor {
+            scheduler: utils::leak(Scheduler::<Event>::new(QueueKind::weights())),
+        }
+    }
+
+    /// Checks and responds to a block validation request.
+    async fn expect_block_validation(
+        &self,
+        expected_block: &ProtoBlock,
+        expected_sender: NodeId,
+        expected_timestamp: Timestamp,
+        valid: bool,
+    ) {
+        let (event, _) = self.scheduler.pop().await;
+        if let Event::BlockValidator(BlockValidationRequest {
+            block,
+            sender,
+            responder,
+            block_timestamp,
+        }) = event
+        {
+            assert_eq!(expected_block, &block);
+            assert_eq!(expected_sender, sender);
+            assert_eq!(expected_timestamp, block_timestamp);
+            responder.respond((valid, block)).await;
+        } else {
+            panic!(
+                "unexpected event: {:?}, expected block validation request",
+                event
+            );
+        }
+    }
+
+    async fn expect_send_message(&self, peer: NodeId) {
+        let (event, _) = self.scheduler.pop().await;
+        if let Event::Network(NetworkRequest::SendMessage {
+            dest, responder, ..
+        }) = event
+        {
+            assert_eq!(peer, dest);
+            responder.respond(()).await;
+        } else {
+            panic!(
+                "unexpected event: {:?}, expected send message request",
+                event
+            );
+        }
+    }
+
+    async fn expect_broadcast(&self) {
+        let (event, _) = self.scheduler.pop().await;
+        if let Event::Network(NetworkRequest::Broadcast { responder, .. }) = event {
```

Comment:
So we're ignoring the broadcast request and pretend that it was executed properly?

Reply:
Yes; this is a unit test: I'm basically ignoring all the other components.